front/src/modules/trainschedule/components/ManageTrainSchedule/UpdateTrainSchedulesV2.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+
+import { Pencil } from '@osrd-project/ui-icons';
+import { useTranslation } from 'react-i18next';
+import { useSelector } from 'react-redux';
+
+import { MANAGE_TRAIN_SCHEDULE_TYPES } from 'applications/operationalStudies/consts';
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { useOsrdConfSelectors } from 'common/osrdContext';
+import { useStoreDataForRollingStockSelector } from 'modules/rollingStock/components/RollingStockSelector/useStoreDataForRollingStockSelector';
+import checkCurrentConfig from 'modules/trainschedule/components/ManageTrainSchedule/helpers/checkCurrentConfig';
+import { setFailure, setSuccess } from 'reducers/main';
+import type { OsrdConfState } from 'reducers/osrdconf/types';
+import { updateSelectedTrainId } from 'reducers/osrdsimulation/actions';
+import { useAppDispatch } from 'store';
+import { formatToIsoDate } from 'utils/date';
+import { castErrorToFailure } from 'utils/error';
+
+import formatTrainSchedulePayload from './helpers/formatTrainSchedulePayload';
+
+type UpdateTrainSchedulesProps = {
+  setIsWorking: (isWorking: boolean) => void;
+  setDisplayTrainScheduleManagement: (arg0: string) => void;
+  setTrainResultsToFetch: (trainScheduleIds?: number[]) => void;
+  trainIdToEdit?: number;
+  setTrainIdToEdit: (trainIdToEdit?: number) => void;
+};
+
+// Refacto in a component to prepare the migration of the patch in editoast (need to be a React component to use hooks inside like rtk's)
+
+const UpdateTrainSchedulesV2 = ({
+  setIsWorking,
+  setDisplayTrainScheduleManagement,
+  setTrainResultsToFetch,
+  trainIdToEdit,
+  setTrainIdToEdit,
+}: UpdateTrainSchedulesProps) => {
+  const { getConf, getName, getStartTime } = useOsrdConfSelectors();
+  const confName = useSelector(getName);
+  const simulationConf = useSelector(getConf);
+  const startTime = useSelector(getStartTime);
+
+  const dispatch = useAppDispatch();
+  const { t } = useTranslation(['operationalStudies/manageTrainSchedule']);
+
+  const [putV2TrainScheduleById] = osrdEditoastApi.endpoints.putV2TrainScheduleById.useMutation();
+
+  const { rollingStock } = useStoreDataForRollingStockSelector();
+
+  async function submitConfUpdateTrainSchedules() {
+    const formattedSimulationConf = checkCurrentConfig(
+      simulationConf as OsrdConfState,
+      t,
+      dispatch,
+      rollingStock?.name
+    );
+
+    if (formattedSimulationConf && trainIdToEdit) {
+      setIsWorking(true);
+      let callSuccess = true;
+      const trainSchedule = formatTrainSchedulePayload(
+        formattedSimulationConf,
+        confName,
+        startTime
+      );
+      try {
+        await putV2TrainScheduleById({
+          id: trainIdToEdit,
+          trainScheduleForm: trainSchedule,
+        })
+          .unwrap()
+          .catch((e) => {
+            callSuccess = false;
+            dispatch(setFailure(castErrorToFailure(e)));
+          });
```

Comment:
I think we can omit this `catch()` call, and the `try`/`catch` block will take care of handling errors?